python/idsse_common/idsse/common/http_utils.py

```diff
@@ -0,0 +1,216 @@
+"""Helper function for listing directories and retrieving s3 objects"""
+# -------------------------------------------------------------------------------
+# Created on Tue Feb 14 2023
+#
+# Copyright (c) 2023 Colorado State University. All rights reserved.             (1)
+# Copyright (c) 2023 Regents of the University of Colorado. All rights reserved. (2)
+#
+# Contributors:
+#     Paul Hamer (1)
+#
+# -------------------------------------------------------------------------------
+
+import logging
+import fnmatch
+import os
+import shutil
+from collections.abc import Sequence
+from datetime import datetime, timedelta, UTC
+
+import requests
+
+from .path_builder import PathBuilder
+from .utils import TimeDelta, datetime_gen
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=duplicate-code, broad-exception-caught
```

Comment:
```suggestion
# pylint: disable=duplicate-code, broad-exception-caught
```
If we have to disable linter we want to have as small of a scope as possible. So 'broad-exception-caught' should be for the specific line of code. I'm guessing that the 'duplicate-code' is because this module is very similar to the aws_utils.py, if that is the case maybe we should make a recourse_utils.py for the common and both aws_utils and http_utils extend that.

Reply:
I raised the possibility of an interface initially, many of the calls made by DAS would be the same for http and aws, at this point it can probably wait for a while.

Reply:
With the refactor we don't need to disable duplicate code, and broad exception should be applied directly to the except and not globally, and ideally we want to specify the specific exception(s) that might happen